packages/portal-gun/virtual.d.ts

```diff
@@ -0,0 +1,27 @@
+type PortalAttrs = { children?: any } & ({ to: string; name: never } | { name: string; to: never });
+
+interface NewElements {
+	portal: PortalAttrs;
+}
+
+declare namespace JSX {
+	export interface IntrinsicElements extends NewElements {}
+}
+
+import 'preact';
+
+declare module 'preact' {
```

Comment:
Why only preact?

Reply:
I added types to React, Preact, Solid, Svelte and Astro.